### PR TITLE
id: lock the Generate function

### PR DESCRIPTION
**Pull request by HunDunDM**
Signed-off-by: Zheng Xiangsheng <hundundm@gmail.com>

### What problem does this PR solve?

* `AllocatorImpl.Alloc` and `AllocatorImpl.Generate` may conflict when executed concurrently.

### What is changed and how it works?

* id: lock the `Generate` function

### Check List

Tests

Related changes

- Need to cherry-pick to the release branch

### Release note

- No release note

### CODE REVIEW

File: server/id/id.go [hunk 1]
```diff
@@ -70,6 +69,13 @@ func (alloc *AllocatorImpl) Alloc() (uint64, error) {
 
 // Generate synchronizes and generates id range.
 func (alloc *AllocatorImpl) Generate() error {
+	alloc.mu.Lock()
+	defer alloc.mu.Unlock()
+
+	return alloc.generateLocked()
+}
+
+func (alloc *AllocatorImpl) generateLocked() error {
```
Comment (Yisaer) [inline server/id/id.go:78]:
```suggestion
func (alloc *AllocatorImpl) unlockedGenerate() error {
```

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

File: server/id/id.go (hunk 1 above)
Comment (HunDunDM) [reply server/id/id.go:78]: `xxxLocked` seems to be the common name in this project, also in `protobuf` or `goleveldb`.

**Review by lhy1024 — APPROVED** [review]

File: server/id/id.go (hunk 1 above)
Comment (JmPotato) [reply server/id/id.go:78]: Also used in the official package such as [sync.Map](https://golang.org/src/sync/map.go).

**Review by JmPotato — APPROVED** [review]

**Pull request merged**